src/Storage/StorageManager.php

```diff
@@ -191,4 +180,137 @@ public function initUploadFile(string $fileHash, int $fileSize, string $fileName
 
         return $count;
     }
+
+    public function addChunk(string $hash, string $chunk, bool $skipShouldSkipServices = true): int
+    {
+        $count = 0;
+        foreach ($this->adapters as $adapter) {
+            if ($adapter->isReadOnly() || ($skipShouldSkipServices && $adapter->shouldSkip())) {
+                continue;
+            }
+            if ($adapter->addChunk($hash, $chunk)) {
+                ++$count;
+            }
+        }
+
+        if ($count === 0) {
+            throw new \RuntimeException(sprintf('Impossible to add a chunk of an asset identified by the hash %s into at least one storage services', $hash));
+        }
+
+        return $count;
+    }
+
+    /**
+     * @return array<string, bool>
+     */
+    public function getHealthStatuses(): array
+    {
+        $statuses = [];
+        foreach ($this->adapters as $adapter) {
+            $statuses[$adapter->__toString()] = $adapter->health();
+        }
+        return $statuses;
+    }
+
+    public function getSize(string $hash): int
+    {
+        foreach ($this->adapters as $adapter) {
+            try {
+                return $adapter->getSize($hash);
+            } catch (\Throwable $e) {
+            }
```

Comment:
empty catch blocks look weird, maybe just continue
```suggestion
            } catch (\Throwable $e) {
                  continue;
            }
```

Reply:
treated in the usage PR